### DISCUSSION TITLE
what/how k3s to do what k8s with Weavenet support multicasting for ROS 2 ?

### DISCUSSION THREAD

**Discussion by swgu931** [k3s-io/k3s#8088] [Q&A]
ROS 2 is the popular robotics platform and it should do message pub/sub on multicasting over same network.
It mean ROS 2 applications running on K3s send/receive message among them by using multicasting
It will work fine in case of all ROS 2 apps running on single node. But it won't work when the one ROS 2 apps running on woker-node-1 and the other running on worker-node-2.

Weavenet CNI on K8s will work fine for any cases.
But Weavenet won't be installed on K3s.  In regards with K3s, how/what we to do with CNI ?

Thanks in advance for your support.

**Comment by brandond** [comment]
> But Weavenet won't be installed on K3s. In regards with K3s, how/what we to do with CNI ?

I'm not sure I understand the question.  Are you stating that weave can't be installed on K3s? Have you tried, and if so, what problems did you run into?

Also, it is my understanding that weave is essentially dead. There are other CNI plugins under active maintenance that do support multicast. You might look into deploying one of those instead.

**Reply by swgu931** [reply]
First of all, I appreciate your comment.

Today we tried to install again on K3s in Ubuntu machine, successfully we fixed the problem with adding the following code in yaml file.
```
- name: IPALLOC_RANGE 
   value: 10.0.0.0/16
```
but it doesn't work at all with the multicasting.

I have been using k8s with weavenet for 3 years after I tried calico, weavenet, flannel and so on. 
And also I search for many reference to find out CNI supporting multicasting. in the end I knew Weavenet is the only one to support multicasting. you can find the paper "Understanding Container Network Interface Plugins: Design Considerations and Performance" and I attach the excerpt below.

![image](https://github.com/k3s-io/k3s/assets/28507261/4cccc5f8-cba0-4a98-bc42-b9f331f662b4)
